website/src/components/explorer/details.jsx

```diff
@@ -0,0 +1,209 @@
+  const digit9 = (split[0] >> 0x12) & THREE_BITS;
+  const digit10 = (split[0] >> 0xf) & THREE_BITS;
+  const digit11 = (split[0] >> 0xc) & THREE_BITS;
+  const digit12 = (split[0] >> 0x9) & THREE_BITS;
+  const digit13 = (split[0] >> 0x6) & THREE_BITS;
+  const digit14 = (split[0] >> 0x3) & THREE_BITS;
+  const digit15 = split[0] & THREE_BITS;
+  return [
+    digit1,
+    digit2,
+    digit3,
+    digit4,
+    digit5,
+    digit6,
+    digit7,
+    digit8,
+    digit9,
+    digit10,
+    digit11,
+    digit12,
+    digit13,
+    digit14,
+    digit15,
+  ];
+}
+
+function cellUnits(hex) {
+  return getResolution(hex) < 8
+    ? { area: UNITS.km2, dist: UNITS.km }
+    : { area: UNITS.m2, dist: UNITS.m };
+}
+
+function ClickableH3Index({ hex, setUserInput }) {
+  const onClick = useCallback(() => {
+    setUserInput(hex);
+  }, [hex, setUserInput]);
+
+  return (
+    <a onClick={onClick} style={{ cursor: "pointer" }}>
+      {hex}
+    </a>
+  );
+}
+
+function ClickableH3IndexList({ hexes, setUserInput, showAll = true }) {
+  const onShowAllClick = useCallback(() => {
+    setUserInput(hexes.join(", "));
+  }, [hexes, setUserInput]);
+  return (
+    <>
+      {hexes.map((hex, index) => {
+        const link = (
+          <ClickableH3Index key={index} setUserInput={setUserInput} hex={hex} />
+        );
+        if (index === 0) {
+          return link;
+        } else {
+          return <span key={index}>, {link}</span>;
+        }
+      })}
+      {showAll ? (
+        <>
+          &nbsp;
+          <a onClick={onShowAllClick} style={{ cursor: "pointer" }}>
+            (show all)
+          </a>
+        </>
+      ) : (
+        <></>
+      )}
+    </>
+  );
+}
+
+export function SelectedHexDetails({
+  setUserInput,
+  splitUserInput,
+  showNavigation = true,
+  showDetails = true,
+}) {
+  if (splitUserInput.length === 1) {
+    const hex = splitUserInput[0];
+    const units = cellUnits(hex);
+    const res = getResolution(hex);
+    const floatPrecision = res / 3 + 7;
+
+    const baseCell = getBaseCellNumber(hex);
+    const pent = isPentagon(hex);
+    const faces = getIcosahedronFaces(hex).join(", ");
+    const coords = cellToLatLng(hex)
+      .map((n) => n.toPrecision(floatPrecision))
+      .join(", ");
+    const digits =
+      res === 0 ? "(none)" : h3IndexToDigits(hex).slice(0, res).join("");
+    const boundary = cellToBoundary(hex);
+    const area = cellArea(hex, units.area).toPrecision(floatPrecision);
+    const edgeLengths = originToDirectedEdges(hex).map((e) =>
+      edgeLength(e, units.dist),
+    );
+    const meanEdgeLength = (
+      edgeLengths.reduce((prev, curr) =>
+        prev !== undefined ? prev + curr : curr,
+      ) / edgeLengths.length
+    ).toPrecision(floatPrecision);
+    const parent = res !== 0 && cellToParent(hex, res - 1);
+    const children = res !== 15 && cellToChildren(hex, res + 1);
+    const neighbors = new Set(gridDisk(hex, 1));
+    neighbors.delete(hex);
```

Comment:
Nit: Maybe `gridDisk(hex, 1).filter(n => n !== hex)` would be simpler, and then you don't need to spread it below